servers/src/main/java/tachyon/master/file/journal/SetStateEntry.java

```diff
@@ -17,32 +17,39 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import tachyon.client.file.options.SetStateOptions;
 import tachyon.master.journal.JournalEntry;
 import tachyon.master.journal.JournalEntryType;
 
-/**
- * This class represents a journal entry for recording the entry of setting a pin.
- */
-public class SetPinnedEntry extends JournalEntry {
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SetStateEntry extends JournalEntry {
   private final long mId;
-  private final boolean mPinned;
   private final long mOpTimeMs;
+  private final Boolean mPinned;
+  private final Long mTTL;
 
-    /**
-     * Creates a new instance of <code>SetPinnedEntry</code>
-     *
-     * @param id the id of the entry
-     * @param pinned whether the entry is pinned or not
-     * @param opTimeMs the operation timestamp (in millisecs)
-     */
+  /**
+   * Creates a new instance of <code>SetStateEntry</code>.
+   *
+   * @param id  the id of the entry
+   * @param opTimeMs the operation timestamp (in millisecs)
+   * @param pinned the pinned flag to be set, otherwise, null
+   * @param ttl the new TTL value to be set, otherwise, null
+   */
   @JsonCreator
-  public SetPinnedEntry(@JsonProperty("id") long id, @JsonProperty("pinned") boolean pinned,
-      @JsonProperty("operationTimeMs") long opTimeMs) {
+  public SetStateEntry(@JsonProperty("id") long id, @JsonProperty("operationTimeMs") long opTimeMs,
+      @JsonProperty("pinned") Boolean pinned, @JsonProperty("ttl") Long ttl) {
     mId = id;
-    mPinned = pinned;
     mOpTimeMs = opTimeMs;
+    mPinned = pinned;
+    mTTL = ttl;
+  }
+
+  public SetStateEntry(long id, long opTimeMs, SetStateOptions options) {
```

Comment:
Do we need this constructor? it is used only in one place in `FileSystemMaster#setState*()` and I would prefer to do the options unwrapping there.